src/editor.c

```diff
@@ -109,6 +109,8 @@ static void close_block(GeanyEditor *editor, gint pos);
 static void editor_highlight_braces(GeanyEditor *editor, gint cur_pos);
 static void read_current_word(GeanyEditor *editor, gint pos, gchar *word, gsize wordlen,
 		const gchar *wc, gboolean stem);
+static void read_current_scope(GeanyEditor *editor, gint pos, gchar *word, gsize wordlen,
```

Comment:
It doesn't read the "current" scope, which is a lexical attribute of the position in the code, rather it reads scope related prefixes that exist in the text.

So I suggest renaming it to `read_scope_prefix`.